### ISSUE TITLE
DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses

### ISSUE THREAD

**Issue by leongross** [Azure/WALinuxAgent#2326]
I created an DCSV2 VM with an Ubuntu 20.04 Gen 2 in azure. I sshed into it and launched my provisioning script. After that I wanted to generalize it to use it as a base image for a shared image gallery definition. I launched this commnd:

```bash
 $ sudo waagent -deprovision
```

And I got the following output: 
```bash
/usr/sbin/waagent:27: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
WARNING! The waagent service will be stopped.
WARNING! Cached DHCP leases will be deleted.
WARNING! root password will be disabled. You will not be able to login as root.
WARNING! /etc/resolv.conf will NOT be removed, this is a behavior change to earlier versions of Ubuntu.
Do you want to proceed (y/n)
```
I enterd y and pressed enter. I got back into my shell and nothing else happend.

Waagent version:
```bash
$ waagent --version
/usr/sbin/waagent:27: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
WALinuxAgent-2.2.46 running on ubuntu 20.04
Python: 3.8.10
Goal state agent: 2.2.46
```

Not log file seems to be generated:
```bash
$ cat /var/log/waagent.log
cat: /var/log/waagent.log: No such file or directory
```

**Comment by leongross** [comment]
UPDATE: When editing the script in /usr/bin/waagent and changing
```python3
import imp
```
to
```python3
import importlib
```
it does not seem to break. Any comments on that?

**Comment by narrieta** [comment]
@leongross This was fixed by #1738, released on 2.2.48.1 (https://github.com/Azure/WALinuxAgent/releases/tag/v2.2.48.1).

It seems like the images published by Canonical are still using 2.2.46. Once they update the images the warning should go away.
